mlir/test/rocmlir-driver/populate_prc.mlir

```diff
@@ -5,17 +5,18 @@
 // F32-NEXT:    call @printMemrefF32(%[[RES]]) : (memref<*xf32>) -> ()
 
 // RUN: rocmlir-gen --arch %arch -p -prc -t f16 | FileCheck %s --check-prefixes=F16,CHECK
-// RUN: rocmlir-gen --arch %arch -p -prc -t bf16 | FileCheck %s --check-prefix=BF16
+// RUN: rocmlir-gen --arch %arch -p -prc -t bf16 | FileCheck %s --check-prefixes=BF16,CHECK
 
 // F16:    %{{.*}} = memref.alloc() : memref<1x128x8x3x3x[[type:f16]]>
-// BF16:   %{{.*}}= memref.alloc() : memref<1x128x8x3x3x[[type:bf16]]>
-// CHECK:  call @_memcpy_[[type]]_f32_9216(%{{.*}}, %{{.*}}) : (memref<9216x[[type]]>, memref<9216xf32>) -> ()
+// BF16:   %{{.*}} = memref.alloc() : memref<1x128x8x3x3x[[type:bf16]]>
+// CHECK:  memref.copy %{{.*}} : memref<9216x[[type]]> to memref<9216x[[type]]>
```

Comment:
Because we now have the test function and the validation function producing the same type, we have memref.copy to directly copy the values rather than memcpy_f16_f32 (for instance) to copy and convert.